packages/contracts-bedrock/scripts/checks/check-interfaces.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+# This script checks for ABI consistency between interfaces and their corresponding contracts.
+# It compares the ABIs of interfaces (files starting with 'I') with their implementation contracts,
+# excluding constructors and certain predefined files. The script reports any differences found
+# and exits with an error if inconsistencies are detected.
+
+# Grab the directory of the contracts-bedrock package
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+CONTRACTS_BASE=$(dirname "$(dirname "$SCRIPT_DIR")")
+
+# Define the files to exclude (patterns can be used)
+EXCLUDE_FILES=(
+  # External dependencies
```

Comment:
`ISchemaRegistry` isn't in `lib`, see https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/EAS/ISchemaRegistry.sol

Reply:
We can't easily ignore `lib` because this is parsing via the `forge-artifacts` folder. We could make the script even smarter by looking at the `compilationTarget` inside of the output JSON but I honestly wouldn't really bother, makes it more brittle.